### PR TITLE
Support usm_ndarray batched input for dpnp.linalg 

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
As noted in #1867 many functions from _dpnp.linalg_ are implemented wrongly for _usm_ndarray_ batched input.
The implementations often call a method that does not support _usm_ndarray_ which raises an error.

This PR suggests fixing this and using _dpnp_ functions instead of _ndarray methods_ and adding a new test to cover these cases

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_linalg.py [hunk 1]
```diff
@@ -57,6 +57,72 @@ def vvsort(val, vec, size, xp):
...
+        pytest.param("tensorinv", {}, {"ind": 1}),
+        pytest.param("tensorsolve", {}, {}),
+    ],
+)
+def test_usm_ndarray_input_batch(func, gen_kwargs, func_kwargs):
+    shape = (
+        (2, 2, 3, 3) if func not in ["tensorinv", "tensorsolve"] else (4, 2, 2)
+    )
+
+    if func in ["lstsq", "solve", "tensorsolve"]:
```
Comment (vtavana) [inline tests/test_linalg.py:88]:
```suggestion

```

File: tests/test_linalg.py [hunk 2]
```diff
@@ -57,6 +57,72 @@ def vvsort(val, vec, size, xp):
...
+                        shape, seed_value=81, **gen_kwargs
+                    )
+                ),
+                dpt.asarray(
+                    generate_random_numpy_array(
+                        shape_b, seed_value=81, **gen_kwargs
+                    )
+                ),
+            ]
+        else:
```
Comment (vtavana) [inline tests/test_linalg.py:103]:
```suggestion
        elif func in ["lstsq", "solve"]:
```

**Review by vtavana — APPROVED** [review]
Thank you, @vlad-perevezentsev!

File: tests/test_linalg.py [hunk 3]
```diff
@@ -57,6 +57,72 @@ def vvsort(val, vec, size, xp):
...
+        pytest.param("matrix_power", {}, {"n": 4}),
+        pytest.param("matrix_rank", {}, {}),
+        pytest.param("norm", {}, {}),
+        pytest.param("pinv", {}, {}),
+        pytest.param("qr", {}, {}),
+        pytest.param("slogdet", {}, {}),
+        pytest.param("solve", {}, {}),
+        pytest.param("svd", {}, {}),
+        pytest.param("tensorinv", {}, {"ind": 1}),
+        pytest.param("tensorsolve", {}, {}),
```
Comment (antonwolfy) [inline tests/test_linalg.py:80]: What is about matrix and vector products? (like tensordot, matmul and so on)
Would you like to cover all linear algebra functions here?

Comment (vlad-perevezentsev) [reply tests/test_linalg.py:82]: It is a good suggestion!
Thanks.
I have added a new test for these functions and also updated them where necessary

**Review by antonwolfy — APPROVED** [review]
Thank you @vlad-perevezentsev

**Pull request merged**